server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1904,7 +1913,8 @@ record RootBlobUpdateResult(RepositoryData oldRepositoryData, RepositoryData new
                         }
                     }));
 
-                    // Write the index metadata for each index in the snapshot
+                    // Write the index metadata and the shard count to memory for each index in the snapshot, so that it persists
+                    // even if the repository is deleted
```

Comment:
"write to memory" is kinda odd phrasing - writes are supposed to be persistent here. Maybe this is leftover from an earlier change?